### PR TITLE
softhsmv2 2.3.0 (new formula)

### PR DESCRIPTION
**Pull request by gijzelaerr**
One warning left:
```
$ brew audit --strict --online softhsmv2
softhsmv2:
  * C: 28: col 26: "#{prefix}/lib" should be "#{lib}"
Error: 1 problem in 1 formula
```
But this is an error in the check I guess.  I'm not doing mkdir `#{prefix}/lib` but mkdir `#{prefix}/var/lib`

### CODE REVIEW

File: Formula/softhsmv2.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Softhsmv2 < Formula
+  desc "SoftHSM version 2"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: Descriptions should not start with the formula name, which this one effectively, if not literally, does.

File: Formula/softhsmv2.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  homepage "http://www.softhsm.org/"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: https://www.opendnssec.org/softhsm/

File: Formula/softhsmv2.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  url "https://github.com/opendnssec/SoftHSMv2/archive/2.3.0.tar.gz"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: https://dist.opendnssec.org/source/softhsm-2.3.0.tar.gz

File: Formula/softhsmv2.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 "9fa32236cb7a344aaea86f44f67a2178e64e8a753073fcf4b88bed437bc1440e"
+
+  depends_on "automake" => :build
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: won't be needed when you use the release tarball

File: Formula/softhsmv2.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  depends_on "autoconf" => :build
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: ditto

File: Formula/softhsmv2.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  depends_on "cppunit" => :build
+  depends_on "libtool" => :build
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: ditto

File: Formula/softhsmv2.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  depends_on "openssl"
+  depends_on "sqlite"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: macOS's sqlite is probably sufficient

File: Formula/softhsmv2.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  def install
+    system "sh", "./autogen.sh"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: won't be needed when you use the release tarball

File: Formula/softhsmv2.rb [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    system "./configure", "--disable-debug",
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: configure: WARNING: unrecognized options: --disable-debug

so it should be removed

File: Formula/softhsmv2.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--with-objectstore-backend-db",
+                          "--with-openssl=/usr/local/opt/openssl",
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: Formula["openssl"].opt_prefix

File: Formula/softhsmv2.rb [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+                          "--with-sqlite3=/usr/local/opt/sqlite",
+                          "--prefix=#{prefix}",
+                          "--sysconfdir=#{etc}"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: `--localstatedir=#{var}`

File: Formula/softhsmv2.rb [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    system "make", "install"
+  end
+
+  test do
+    mkdir "#{prefix}/var/lib/softhsm/tokens/"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: if this path must exist,
```
  def post_install
    (var/"lib/softhsm/softhsm").mkpath
  end
```

File: Formula/softhsmv2.rb [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    system "#{bin}/softhsm2-util", "--init-token", "--free", "--label", "test", "--so-pin", "1234", "--pin", "1234"
```
Comment (ilovezfs) [inline Formula/softhsmv2.rb]: wrap arguments to the next line so it doesn't go passed the eightieth column

**Review by ilovezfs — CHANGES_REQUESTED** [review]
It looks like the formula is installing `libsofthsm2.so` not `libsofthsm2.dylib`, which is a porting issue that needs to be fixed upstream before we can include this software in Homebrew.

File: Formula/softhsm2.rb [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+  depends_on "cppunit" => :build
+  depends_on "openssl"
+
+  def install
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--with-objectstore-backend-db",
+                          "--with-migrate",
+                          "--with-openssl=" + Formula["openssl"].opt_prefix,
```
Comment (DomT4) [inline Formula/softhsm2.rb]:
```ruby
"--with-openssl=#{Formula["openssl"].opt_prefix}",
```

File: Formula/softhsm2.rb [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+                          "--prefix=#{prefix}",
+                          "--sysconfdir=#{etc}",
+                          "--localstatedir=#{var}"
+
+    system "make", "install"
+
+    include.install "src/lib/pkcs11/cryptoki.h",
```
Comment (ilovezfs) [inline Formula/softhsm2.rb]: Why are we installing these if make install doesn't? That's an upstream bug that needs to be fixed if they "should" be installed but aren't.

File: Formula/softhsm2.rb [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+                    "src/lib/pkcs11/pkcs11f.h",
+                    "src/lib/pkcs11/pkcs11t.h"
+  end
+
+  def post_install
+    (var/"lib/softhsm/softhsm").mkpath
+  end
+
+  test do
+    system "#{bin}/softhsm2-util", "--init-token", "--free",
```
Comment (ilovezfs) [inline Formula/softhsm2.rb]: The test needs to do something other than check for a 0 exit code here.

File: Formula/softhsm2.rb (hunk 15 above)
Comment (gijzelaerr) [reply Formula/softhsm2.rb]: checked upstream, yes we should not install them

File: Formula/softhsm2.rb (hunk 16 above)
Comment (gijzelaerr) [reply Formula/softhsm2.rb]: what do you recommend? I have no idea what else I could do.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/softhsm2.rb [hunk 17]
```diff
@@ -0,0 +1,33 @@
+class Softhsm2 < Formula
+  desc "Cryptographic store accessible through a PKCS #11"
+  homepage "https://www.opendnssec.org/softhsm/"
+  url "https://dist.opendnssec.org/source/softhsm-2.3.0.tar.gz"
+  sha256 "5ed604c89a3a6ef9d7d1ee92c28a2c4b3cd1f86f302c808e2d12c8f39aa2c127"
+
+  depends_on "cppunit" => :build
```
Comment (ilovezfs) [inline Formula/softhsm2.rb]: can we disable the tests instead?

Comment (ilovezfs) [reply Formula/softhsm2.rb]: actually nevermind I see this is security software so it's fine

**Review by ilovezfs — CHANGES_REQUESTED** [review]
I don't think we need the `2` on the formula name

File: Formula/softhsm2.rb (hunk 16 above)
Comment (ilovezfs) [reply Formula/softhsm2.rb]: I'm getting
```
bash-3.2$ /usr/local/Cellar/softhsm2/2.3.0/bin/softhsm2-util --init-token --free --label test --so-pin 1234 --pin 1234
Slot 0 has a free/uninitialized token.
ERROR rv=0x00000030: Could not initialize the token.
Please check log files for additional information.
bash-3.2$ echo $?
1
bash-3.2$
```
So not sure it's working as expected right now.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
```
Testing softhsm2
/usr/bin/sandbox-exec -f /tmp/homebrew20171201-31441-k30obm.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/softhsm2.rb --verbose
==> /usr/local/Cellar/softhsm2/2.3.0/bin/softhsm2-util --init-token --free --label test --so-pin 1234 --pin 1234
ERROR rv=0x00000030: Could not initialize the token.
Please check log files for additional information.
Slot 0 has a free/uninitialized token.
Error: softhsm2: failed
```

**Comment by ilovezfs** [comment]
Please feel free to re-open when it's building a .dylib not a .so and the test works. Thanks for the PR @gijzelaerr

**Comment by bellgrim** [comment]
@ilovezfs PKCS#11 should be considered as dynamically loadable modules (.so/.bundle) and not as shared libraries (.dylib). That you dynamically load (dlopen) and not via dynamic linking.

Compare with another PKCS#11 library: http://brewformulas.org/Opensc

They even reverted from .dylib to .so: https://github.com/OpenSC/OpenSC/commit/630e7ee08efe01c4a8db621df373f19153f8a2fd#diff-67e997bcfdac55191033d57a16d1408a

**Pull request closed without merging**